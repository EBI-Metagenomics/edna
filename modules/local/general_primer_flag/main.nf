
process GENERAL_PRIMER_FLAG {
    // Check for the presence of primers in general
    tag "$meta.id"
    label 'very_light'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        "https://depot.galaxyproject.org/singularity/mi-pimento:1.0.0--pyhdfd78af_0":
        "biocontainers/mi-pimento:1.0.0--pyhdfd78af_0"}"
    input:
    tuple val(meta), path(reads_merged)

    output:
    tuple val(meta), path("*general_primer_out.txt"), emit: general_primer_out
    path "versions.yml"                             , emit: versions

    script:
    """
    pimento are_there_primers -i ${reads_merged} -o ${meta.id}_${meta.var_region}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mi-pimento: \$( pimento --version | cut -d" " -f3 )
    END_VERSIONS
    """
}
