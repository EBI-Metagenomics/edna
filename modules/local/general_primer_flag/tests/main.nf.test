nextflow_process {

    name "Test Process GENERAL_PRIMER_FLAG"
    script "../main.nf"
    process "GENERAL_PRIMER_FLAG"
    tag "modules"
    tag "general_primer_flag"

    test("Should take a fastq.gz file and try to identify whether primers exist in general, outputting a .txt file containing true/false values for both directions") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false, var_region:'16S-V3-V4' ], // meta map
                    file('${projectDir}/modules/local/std_primer_flag/tests/data/test.fastq.gz', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.general_primer_out).match("general_primer_out") }
            )
        }

    }

}
