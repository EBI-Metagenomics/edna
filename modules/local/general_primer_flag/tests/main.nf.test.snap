{
    "general_primer_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false,
                        "var_region": "16S-V3-V4"
                    },
                    "test_16S-V3-V4_general_primer_out.txt:md5,f2160c8ffedf48068f2e1137e0a3a7e7"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-25T11:29:39.559975639"
    }
}